guide/src/building_and_distribution.md

```diff
@@ -1,58 +1,91 @@
 
-[features]
-default = ["pyo3/extension-module"]
-```
+There are also [`maturin-starter`] and [`setuptools-rust-starter`] examples in the PyO3 repository.
 
-And this in the profiling crate:
+### Manual builds
 
-```toml
-[dependencies]
-my_main_crate = { path = "..", default-features = false }
-pyo3 = { {{#PYO3_CRATE_VERSION}} }
-```
+To build a PyO3-based Python extension manually, start by running `cargo build` as normal in a library project which uses PyO3's `extension-module` feature and has the [`cdylib` crate type](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#the-crate-type-field).
 
-On Linux/macOS you might have to change `LD_LIBRARY_PATH` to include libpython, while on windows you might need to set `LIB` to include `pythonxy.lib` (where x and y are major and minor version), which is normally either in the `libs` or `Lib` folder of a Python installation.
+Once built, symlink (or copy) and rename the shared library from Cargo's `target/` directory to your desired output directory:
+- on macOS, rename `libyour_module.dylib` to `your_module.so`.
+- on Windows, rename  `libyour_module.dll` to `your_module.pyd`.
+- on Linux, rename `libyour_module.so` to `your_module.so`.
 
-## Testing, building and distribution
+You can then open a Python shell in the output directory and you'll be able to run `import your_module`.
 
-There are two main ways to test, build and distribute your module as a Python package: [setuptools-rust] and [maturin]. setuptools-rust needs several configuration files (`setup.py`, `MANIFEST.in`, `build-wheels.sh`, etc.). It allows (and sometimes requires) writing custom workflows in python. maturin has only few options and works without any additional configuration, instead it requires a rigid project structure and does not support some functionality of setuptools such as package data ([pyo3/maturin#258](https://github.com/PyO3/maturin/issues/258)), multiple extensions or running python scripts at build time.
+See as an example, Bazel rules to build PyO3 on linux at https://github.com/TheButlah/rules_pyo3
 
-### Manual builds
+### The `extension-module` feature
 
-You can also symlink (or copy) and rename the shared library from the `target` folder:
-- on macOS, rename `libyour_module.dylib` to `your_module.so`.
-- on Windows, rename  `libyour_module.dll` to `your_module.pyd`.
-- on Linux, rename `libyour_module.so` to `your_module.so`.
+PyO3's `extension-module` feature is used to disable [linking](https://en.wikipedia.org/wiki/Linker_(computing)) to `libpython` on unix targets.
 
-You can then open a Python shell in the same folder and you'll be able to run `import your_module`.
+This is necessary because by default PyO3 links to `libpython`. This makes binaries, tests, and examples "just work". However, Python extensions on unix must not link to libpython for [manylinux](https://www.python.org/dev/peps/pep-0513/) compliance).
```

Comment:
```suggestion
This is necessary because by default PyO3 links to `libpython`. This makes binaries, tests, and examples "just work". However, Python extensions on unix must not link to libpython for [manylinux](https://www.python.org/dev/peps/pep-0513/) compliance.
```